app/src/main/java/org/gophillygo/app/data/models/Destination.java

```diff
@@ -76,10 +79,9 @@ public Destination(int id, int placeID, String name, boolean accessible, String
         this.categories = categories;
     }
 
-    public void setDistance(float distance) {
+    public void setDistance(Context context, float distance) {
```

Comment:
Without having a `public void setDistance(float distance) { ... }` method defined I can't get the project to compile.

I think a good way to fix this is might be to defer loading the resource until needed by changing `getFormattedDistance` to take a `Context` parameter. There is an implicit `context` variable according to S.O. https://stackoverflow.com/questions/32374483/android-databinding-where-to-get-context

Reply:
Adding a separate context-free public setter to satisfy Gson/Room is all that's needed. The build error didn't show until after a cache invalidation.

Deferring to the getter would defeat the advantage of only calculating it once and storing it, and there's context already available when and where it gets set.

Reply:
https://github.com/azavea/cac-tripplanner-android/pull/116/commits/d76f73f97d991d9d487af9aee9d3c12f0d2c3a5a allowed the app to build, but now destinations do not show distance:

![screenshot_20180703-145918](https://user-images.githubusercontent.com/4432106/42239574-5650eda8-7ed2-11e8-99c3-76e9e46f6c2f.png)

Note if I use the up navigation to return to the home activity the formatted distance actually shows up for short while but then is hidden.

Reply:
Yeah, looked like it should work, but silently fails to set anything. I'll go with the modifications to format on read instead.